src/bridge.ts

```diff
@@ -119,10 +121,12 @@ export class Bridge extends EventEmitter {
 
   public async initBrowser (): Promise<Browser> {
     log.verbose('PuppetPuppeteerBridge', 'initBrowser()')
-
+    const launchOptions = (this.options.launchOptions as LaunchOptions) || {}
     const headless = !(this.options.head)
-    const executablePath = this.options.endpoint
+    const executablePath = this.options.endpoint || launchOptions.executablePath
```

Comment:
Move `endpoint` to `executablePath` inside `launchOptions` to prevent some race conditions and unpredicatable results:

```diff
-    const executablePath = this.options.endpoint || launchOptions.executablePath
+  if (this.options.endpoint) {
+    launchOptions.executablePath = this.options.endpoint
+  }
```